drivers/HwWatchdog.cpp

```diff
@@ -0,0 +1,84 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "HwWatchdog.h"
+
+namespace mbed {
+rtos::Mutex HwWatchdog::_Mutex;
+mbed_error_status_t HwWatchdog::start()
+{
+
+     watchdog_status_t sts;
+     MBED_ASSERT(HW_WATCHDOG_TIMEOUT != 0);
+
+     MBED_ASSERT(HW_WATCHDOG_TIMEOUT < max_timeout());
+
+    _Mutex.lock();
+    watchdog_config_t config;
+    config.timeout_ms = HW_WATCHDOG_TIMEOUT;
+
+    sts = hal_watchdog_init(&config);
+    if(sts != WATCHDOG_STATUS_OK ) {
+        _Mutex.unlock();
+        return MBED_ERROR_INITIALIZATION_FAILED;
+     }
+    float timeout = (HW_WATCHDOG_TIMEOUT /2);
+    _ticker.attach(callback(this,&HwWatchdog::kick),(timeout / 1000));
+
+    _Mutex.unlock();
+    return MBED_SUCCESS;
+}
+
+
+void HwWatchdog::kick()
+{
+    hal_watchdog_kick();
+}
+
+
+mbed_error_status_t HwWatchdog::stop()
+{
+    watchdog_status_t sts;
+    _Mutex.lock();
+    sts = hal_watchdog_stop();
+    if(sts != WATCHDOG_STATUS_OK ) {
+      _Mutex.unlock();
+      return MBED_ERROR_OPERATION_PROHIBITED;
+    }
+    _Mutex.unlock();
+    return MBED_SUCCESS;
+}
+
+
+uint32_t HwWatchdog::reload_value() const
+{
+    return hal_watchdog_get_reload_value();
+}
+
+
+uint32_t HwWatchdog::max_timeout()
+{
+    const watchdog_features_t features = hal_watchdog_get_platform_features();
```

Comment:
Are we returning a structure in return value?

Reply:
Returning structure from hal API? Not sure how this was missed
https://github.com/ARMmbed/mbed-os/blob/feature-watchdog/hal/watchdog_api.h#L137 

CC @donatieng @c1728p9

Reply:
we are returning only the max_timeout